packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -934,14 +965,14 @@ const findMentionTagForSelection = (tags: TagData[], selection: number): TagData
         const selectedTag = findMentionTagForSelection(tag.subTags, selection);
         if (selectedTag !== undefined) {
           mentionTag = selectedTag;
-          break;
+          return;
```

Comment:
I think we should return selectedTag here?

Reply:
Indeed, the logic here needs updating.